lib/template-loader.php

```diff
@@ -383,3 +383,26 @@ function gutenberg_template_loader_filter_block_editor_settings( $settings ) {
 	return $settings;
 }
 add_filter( 'block_editor_settings', 'gutenberg_template_loader_filter_block_editor_settings' );
+
+/**
+ * Removes post details from block context when rendering a block template.
+ *
+ * @param array $context Default context.
+ *
+ * @return array Filtered context.
+ */
+function gutenberg_template_render_without_post_block_context( $context ) {
+	/*
+	 * Blocks rendered for a template should be treated as if there is no
+	 * current post, since the purpose of a template is in prescribing how
+	 * blocks appear for any post which matches the template. The post is
+	 * assigned only once the template matches in template loading.
```

Comment:
>The post is
	 * assigned only once the template matches in template loading.

What does that mean?

Reply:
> > The post is
> > * assigned only once the template matches in template loading.
> 
> What does that mean?

The intention was to describe how a post is determined only at the time of template rendering. In retrospect, I can see how the wording is confusing. I tried a different take in the rebased 1c5d9e4e010ecdefcde954f43a53b4458735ca7c .

>The post is assigned only once the template is used in rendering a post.

Although even now I see the repetition of "post" is maybe non-ideal 😅 Does it make sense? Any suggestions?

Reply:
>When loading a template or template part directly and not through a page that resolves it, the top-level post ID and type context get set to the template part's. Templates are just the structure of a site, and they should not be available as post context because blocks like Post Content would recurse infinitely.

Reply:
> > When loading a template or template part directly and not through a page that resolves it, the top-level post ID and type context get set to the template part's. Templates are just the structure of a site, and they should not be available as post context because blocks like Post Content would recurse infinitely.

Updated in e2c84a3.